File: 11-19/youi-1119-p1.cpp
#include <iostream>
#include <vector>
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
using namespace std;


struct ListNode {
      int val;
      ListNode *next;
      ListNode(int x) : val(x), next(NULL) {}
};

class Solution {
public:
    vector<int> reversePrint(ListNode* head) {
        ListNode* node = head;
        vector<int> res;

        while(node){
            res.insert(res.begin(),node->val);
            node= node->next;
        }

        return res;
    }
};

File: day2/youi-1119-p1.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def reversePrint(self, head: ListNode) -> 'list[int]':
        res = []
        node = head
        if not node :
            return []
        while node:
            res.insert(0,node.val)
            node = node.next
        return res
        